### ISSUE TITLE
Instrument user lifecycle stage

### ISSUE THREAD

**Issue by Firewall** [containers/podman-desktop#10329]
### Is your feature request related to a problem? Please describe

It would be useful to now what stage that the user is in after the onboarding so we can make improvements to our onboarding where needed. 

There are 2 stages that would be useful to know.

1. `If the user has a container provider.` This will help us to know if the user not only made it through the onboarding, but that the user also succesfully was able to create a provider (eg. a podman machin) during it.
2. `If the user has containers`. This helps us to know that the user not only has the ability to run a conatiner, but actually was succesful in doing that.

### Describe the solution you'd like

When the user opens Podman Desktop track the following
- [ ]  If the user has a container provider
- [ ] The amount of containers the user has

### Describe alternatives you've considered

There are other places we can add this event. This was the first idea that was discussed with @benoitf but discuss if there are other ideas.

**Comment by jeffmaury** [comment]
The issue as described as little relation with onboarding. I think it should be reworked to better understand the needs based on the following inputs:

- onboarding already has telemetry, do we have something missing from the existing telemetry
- the information specified above is not related to onboarding so is there really a need to get it

**Comment by Firewall** [comment]
@jeffmaury Yep the onboarding already has telemetry, but when a user upgrades from a previous version, or opens the app on day 2 they don't go through the onboarding again. The onboading telemetry is not triggering at that point. I changed the title of the ticket because you are right its more tracking in what stage of the user lifecycle the user is. 

The way this data connects to the onboarding is that during the onboading we are helping the user to:
1. Set up the Podman CLI, if they don't have it
2. Create a Podman Machine if they don't have one

We can tell from telemetry if they were successful at that whilst they go through the onboarding. We already have dashboards where we can see succes rates of that. 

However many users (most) are not using Podman Desktop for the very first time, and thus won't see the onboarding. When they open the Podman Desktop app, they will load the dashboard. These metrics is understand how those users are doing. "Do they have a running machine?" & 'Are they able to launch containers"?

**Comment by axel7083** [comment]
> These metrics is understand how those users are doing. "Do they have a running machine?" & 'Are they able to launch containers"?

I spent some time in amplitude, you want to know how many machines does a user have on startup. You may look at the `registerContainerProviderConnection` event.

https://github.com/podman-desktop/podman-desktop/blob/d7e1f5065e883396c3d87c16dc3c549b566a89b3/packages/main/src/plugin/container-registry.ts#L327-L331

This event is collected every time a a podman machine is registered (usually at startup). 

To be able to have a ratio, let's say _How many podman machines does user have?_ we should take the number of users using the podman extension

Here is an amplitude chart https://app.amplitude.com/analytics/redhat/chart/q41876dy/edit/aijb1d5a that display this ratio.

**Comment by axel7083** [comment]
As discussed with @jeffmaury closing as we already have the information needed collected
